ntp-proto/src/nts_record.rs

```diff
@@ -1309,19 +1382,40 @@ impl KeyExchangeServer {
                                 let algorithm = result.algorithm;
                                 let protocol = result.protocol;
 
-                                tracing::debug!(?algorithm, "selected AEAD algorithm");
+                                tracing::debug!(?algorithm, "selected AEAD algorithm",);
+
+                                #[cfg(feature = "nts-pool")]
+                                let keys = if let Some(keys) = result.fixed_keys {
+                                    if self.privileged_connection() {
+                                        tracing::debug!("using fixed keys for AEAD algorithm");
+                                        algorithm
+                                            .try_into_nts_keys(keys)
+                                            .ok_or(KeyExchangeError::InvalidFixedKeyLength)
+                                    } else {
+                                        tracing::debug!("refused fixed key request due to improper authorization");
+                                        Err(KeyExchangeError::UnrecognizedCriticalRecord)
+                                    }
+                                } else {
+                                    tracing::debug!(
+                                        "using AEAD keys extracted from TLS connection"
+                                    );
+                                    algorithm
+                                        .extract_nts_keys(protocol, &self.tls_connection)
+                                        .map_err(KeyExchangeError::Tls)
+                                };
 
-                                let keys = match algorithm
+                                #[cfg(not(feature = "nts-pool"))]
+                                let keys = algorithm
                                     .extract_nts_keys(protocol, &self.tls_connection)
-                                {
-                                    Ok(keys) => keys,
-                                    Err(e) => {
-                                        return ControlFlow::Break(Err(KeyExchangeError::Tls(e)))
-                                    }
+                                    .map_err(KeyExchangeError::Tls);
+
+                                let send_response = || -> Result<(), KeyExchangeError> {
+                                    self.send_response(protocol, algorithm, keys?)
+                                        .map_err(KeyExchangeError::Io)
                                 };
```

Comment:
Why the indirect control flow here? I would recommend we rewrite this to just storing the result in a variable before doing the match.

Reply:
This is an attempt at making the matching level that was present here less horrible to read (in lieu of try blocks).

Reply:
But I can make something with `Result::and_then` that should satisfy you.